packages/flutter/lib/src/widgets/form.dart

```diff
@@ -371,6 +430,7 @@ class FormFieldState<T> extends State<FormField<T>> {
   void reset() {
     setState(() {
       _value = widget.initialValue;
+      _hasInteractedByUser = false;
```

Comment:
we may need this here
```dart
Form.of(context)?._fieldDidChange();
```

Reply:
I removed it because it wan't there before.

Since it is not required for this feature I though that it is better to remove it for now and add in the next PR when making performance improvements.

Reply:
This not a performance improvement but a corner case i just discovered. A field reset need to notify the form to recalculate its _hasInteractedByUser and rebuild. otherwise,  the form will keep validating all its fields.

Reply:
Done